tests/test_installer.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2024 Science project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import subprocess
+from pathlib import Path
+
+import pytest
+from _pytest.tmpdir import TempPathFactory
+
+from science.os import IS_WINDOWS
+
+
+@pytest.fixture(scope="module")
+def installer(build_root: Path) -> list:
+    installer = build_root / "install.sh"
+    return [Path(r"C:\Program Files\Git\bin\bash.EXE"), installer] if IS_WINDOWS else [installer]
```

Comment:
This obviously works with the GH runners but I'll probably circle back and either plumb the bash executable path / specialize in CI or else just implement the PowerShell equivalent 1st class. I looked at examples from uv, deno and bun and there is not too much PowerShell to learn.

Reply:
hmm, I think just checking if the git-bash path exists first - then if not falling back to "bash.exe" (which did work initially, but gave an error about WSL not having the modules installed to run bash etc) - should provide unilateral support for the "windows developer that has git bash and/or WSL installed" env assumption?

either way, feel free to iterate accordingly - esp if there are broader cases of a need for discovering a usable bash for testing etc.

Reply:
Yeah, you mistook my meaning. I just mean to do something like this in the CI workflow step that runs tests:
```yaml
run: |
  export _PEX_TEST_USE_BASH=$SHELL
  ... rest ...
```

Then in the test, use that env var if set.

Reply:
Basically, by trying to do this test using pytest, you inherit the extra burden of having to simulate the user env. vs just running the install.sh script via CI bash / bash tests.

Reply:
gotcha. I was definitely thinking about both CI and developers running these tests on Windows.